java/org/apache/coyote/http2/Stream.java

```diff
@@ -604,10 +611,23 @@ final void writeAck() throws IOException {
     final void writeEarlyHints() throws IOException {
         MimeHeaders headers = coyoteResponse.getMimeHeaders();
         String originalStatus = headers.getHeader(":status");
-        headers.setValue(":status").setString("103");
+        headers.setValue(":status").setString(String.valueOf(HttpServletResponse.SC_EARLY_HINTS));
+
         try {
-            handler.writeHeaders(this, headers, false, Constants.DEFAULT_HEADERS_FRAME_SIZE);
+            MimeHeaders earlyHintsHeaders = new MimeHeaders();
+            earlyHintsHeaders.duplicate(headers);
+            earlyHintsHeaders.filter(HTTP_EARLY_HINTS_HEADERS);
```

Comment:
This isn't how early hints are meant to work. The headers sent via early hints are expected to be in the final response.

Reply:
I am well aware of all of that. That is all an application concern, not a container concern. My previous comments stand.

Reply:
`HttpServletResponse.setHeader(String,String)` to remove (set to `null`) or replace (set to new value).